src/sleep/sleep.c

```diff
@@ -316,10 +313,10 @@ static int custom_timer_suspend(const SleepConfig *sleep_config) {
                 woken_by_timer = FLAGS_SET(r, POLLIN);
 
                 r = fetch_batteries_capacity_by_name(&current_capacity);
-                if (r < 0) {
+                if (r < 0 || hashmap_isempty(current_capacity)) {
```

Comment:
feels like this change should be in its own commit, with a brief explanation? doesn't seem related to ENOENT